authentication/src/main/kotlin/com/uber/sdk2/auth/api/exception/AuthException.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (C) 2024. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.uber.sdk2.auth.api.exception
+
+/** Represents the exception that occurred during the authentication request. */
+sealed class AuthException(override val message: String) : Throwable(message) {
```

Comment:
I think we should not extend from `Throwable`. It is the top type and encompasses all `Error` which are always unrecoverable (e.g. `StackOverflowError`.

Should we extend from `RuntimeException`?

Reply:
Good call